The various curl NTLM tests try to LD_PRELOAD a test library
called .../tests/libtest/.libs/libhostname.so that's been
created as a pre-requisite to running the numerous Perl tests.

This patch adjusts LD_PRELOAD to be LD_PRELOAD_32 or LD_PRELOAD_64
depending upong whether we are running the 32 or 64 bit tests.
This is required because during testing are the compiled binaries
executed via shell wrapper. The shell in shebang line is #!/bin/sh which
is 64bit binary and fails when you try to preload 32bit binary via
LD_PRELOAD.

As this is Solaris specific, this patch will not be pushed upstream.

--- curl-7.62.0/tests/runtests.pl   2018-11-08 03:22:37.095822288 +0000
+++ curl-7.62.0/tests/runtests.pl   2018-11-08 03:16:26.092582609 +0000
@@ -3651,6 +3651,12 @@ sub singletest {
                             # print "Skipping LD_PRELOAD due to no release shared build\n";
                             next;
                         }
+                        # make this LD_PRELOAD_{bits}
+                        open(FP, "/bin/file $content |");
+                        my $bits = <FP>;
+                        if ($bits =~ /^.+ELF\s(\d\d)-bit.+$/) {
+                            $var .= '_'.$1;
+                        }
                     }
                     $ENV{$var} = "$content";
                 }
